lib/form/js/repeatable.js

```diff
@@ -4,257 +4,338 @@
+      }
+      else if (child.htmlFor && child.htmlFor !== '') {
+        child.htmlFor = splitPop(child.htmlFor, '--');
+      }
+    });
+  };
+
+  var deleteHandler = function deleteEventHandler(e) {  // eslint-disable-line one-var, func-names
+    var node = e.currentTarget.parentNode,
+        siblingNode = node.nextElementSibling,
+        container,
+        parent = node.parentNode,
+        attribute = parent.id;
+
+    // removes child and update length
+
+    parent.removeChild(node);
+    allRepeatables[attribute].length--;
+    deleteIdsAndInputs(node.querySelectorAll('*'));
+
+    // change ids and names of subsequest instances
+    while (siblingNode) {
+      if (siblingNode.getAttribute('class') === 'form--repeatable') {
+        updateInstance(siblingNode);
+      }
+      siblingNode = siblingNode.nextElementSibling;
+    }
+
+    // remove index if there is just one instance left
+    if (allRepeatables[attribute].length === 1) {
+      container = parent.querySelector('.form--repeatable');
+      removeIndex(container);
+      container.removeChild(container.querySelector('.form--delete'));
+    }
+  };
+
+  // Add new input to allInputs
+  var setIDsAndInputs = function (idMap) {  // eslint-disable-line one-var, func-names
+    Object.keys(idMap).forEach(function setIdandInput(id) {
+      if (allInputs.hasOwnProperty(id)) {
+        allInputs[idMap[id]] = Object.assign({}, allInputs[id]);
+        if (!allIDs.hasOwnProperty(idMap[id])) {
+          allIDs[idMap[id]] = document.getElementById(idMap[id]);
+
+          // Adds validation to new input
+          allIDs[idMap[id]].addEventListener(allInputs[idMap[id]].validation.on, validate);
         }
-      });
+      }
+    });
+  };
 
-      // Increments length of attribute
-      allRepeatables[input].length++;
-      containerSet = container.parentNode.querySelectorAll('.form--repeatable');
+  // Create and return delete button
+  var newDeleteButton = function (attribute) {  // eslint-disable-line one-var, func-names
+    var element = document.createElement('button');
+    element.type = 'button';
+    element.textContent = 'Delete';
+    element.id = attribute + '--delete';
+    element.setAttribute('class', 'form--delete');
+    element.addEventListener('click', deleteHandler);
 
-      // Retrives element next to last container
-      lastContainer = containerSet[containerSet.length - 1].nextSibling;
+    return element;
+  };
+
+  // Update values of current instance
+  var updateCurrentInstance = function (instance, attribute) {  // eslint-disable-line one-var, func-names
+    var container = instance,
+        index = allRepeatables[attribute].length;
+
+    // Adds new delete button to container
+    container.appendChild(newDeleteButton(attribute));
+
+    // Adds index to id
+    container.id = container.id + '--' + (index - 1);
+
+    // Updates all children of container
+    container.querySelectorAll('*').forEach(function setAttributes(node) {
```

Comment:
Did you test this across browsers? [`querySelectorAll`](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelectorAll) returns a [`NodeList`](https://developer.mozilla.org/en-US/docs/Web/API/NodeList) which is _not_ an Array and does not have `forEach` available. In browsers that implement `NodeList` properly, this will break.

You should either loop over these with a `for` loop, or use `Array.prototype.forEach.call()` instead